suite/compatibility/json_rpc_test.go

```diff
@@ -0,0 +1,140 @@
+}
+
+type Method struct {
+	Name    string        `json:"name"`
+	Summary string        `json:"summary"`
+	Params  []interface{} `json:"params"`
+	Result  Result        `json:"result"`
+}
+
+type OpenRPCStruct struct {
+	Openrpc string      `json:"openrpc"`
+	Info    interface{} `json:"info"`
+	Methods []Method    `json:"methods"`
+}
+
+type RPCMethodCalls map[string][]interface{}
+
+var rpcMethodCalls RPCMethodCalls
+
+var _ = Describe("Json RPC compatibility @json_rpc", func() {
+	var (
+		openrpcSchema OpenRPCStruct
+		methods       []Method
+	)
+	rpcClientsByChain := make(map[int][]*rpc.Client)
+
+	jsonFile, err := os.Open(filepath.Join(utils.ProjectRoot, "openrpc.json"))
+	Expect(err).ShouldNot(HaveOccurred())
+	defer func(jsonFile *os.File) {
+		_ = jsonFile.Close()
+	}(jsonFile)
+
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	err = json.Unmarshal(byteValue, &openrpcSchema)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	methods = openrpcSchema.Methods
+
+	rpcMethodCalls = RPCMethodCalls{
+		"eth_getBlockByNumber": []interface{}{"0x000000"},
+		"eth_chainId":          []interface{}{},
+		"eth_gasPrice":         []interface{}{},
+		"eth_getBalance":       []interface{}{"0x0000000000000000000000000000000000000000"},
+		"eth_getCode":          []interface{}{"0x0000000000000000000000000000000000000000"},
+		"eth_getLogs": []interface{}{map[string]interface{}{
+			"fromBlock": "0x000000",
+			"toBlock":   "0x000010",
+			"address":   "0x0000000000000000000000000000000000000000",
+		}},
+	}
+
+	BeforeEach(func() {
+		By("Getting RPC clients", func() {
+			nc, err := config.LoadNetworksConfig(filepath.Join(utils.ProjectRoot, "networks.yaml"))
+			Expect(err).ShouldNot(HaveOccurred())
+			for _, networkName := range nc.SelectedNetworks {
+				networkSettings, ok := nc.NetworkSettings[networkName]
+				Equal(ok)
+				chainId := networkSettings["chain_id"].(int)
+				urls := networkSettings["urls"]
+				var rpcClients []*rpc.Client
+				for _, url := range urls.([]interface{}) {
+					rpcClient, err := rpc.Dial(fmt.Sprintf("%v", url))
+					Expect(err).ShouldNot(HaveOccurred())
+					rpcClients = append(rpcClients, rpcClient)
+				}
+				rpcClientsByChain[chainId] = rpcClients
+			}
+		})
+	})
+
+	Describe("Test GET RPC methods and validate results", func() {
+		It("OCR test GET Methods", func() {
+			for chainId, rpcClients := range rpcClientsByChain {
+				fmt.Printf("Starting tests for chain ID %d\n", chainId)
+				for rpcMethod, rpcMethodParameters := range rpcMethodCalls {
+					fmt.Printf("\n\nMethod: %s", rpcMethod)
+					var method Method
+					for _, value := range methods {
+						if value.Name == rpcMethod {
+							method = value
+							break
+						}
+					}
+					schemaLoader := gojsonschema.NewGoLoader(method.Result.Schema)
+					for _, rpcClient := range rpcClients {
+						var rpcCallResult interface{}
+						err := rpcClient.CallContext(context.Background(), &rpcCallResult, rpcMethod, rpcMethodParameters...)
+						if err != nil {
```

Comment:
Make use of the Ginkgo matchers and assertions instead.

Reply:
I want test to continue so I don't use gomega assertions and use logging instead.
I'm not sure if it is reasonable to make separate tests for each JSON RPC method and use assertion for them.
If it is possible to continue test after failed assertion I will use it.